lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+        """
+
+        changed = False
+        if "newName" in self.data:
+            self.data["name"] = self.data.pop("newName")
+
+        if not resource:
+            resource = getattr(self.resource_client, create_method)(self.data)
+            msg = self.MSG_CREATED
+            changed = True
+
+        else:
+            merged_data = resource.copy()
+            merged_data.update(self.data)
+
+            if ResourceComparator.compare(resource, merged_data):
+                msg = self.MSG_ALREADY_PRESENT
+            else:
+                resource = self.resource_client.update(merged_data)
+                changed = True
+                msg = self.MSG_UPDATED
+
+        return dict(
+            msg=msg,
+            changed=changed,
+            ansible_facts={fact_name: resource}
+        )
+
+    @staticmethod
+    def transform_list_to_dict(list_):
+        """
+        Transforms a list into a dictionary, putting values as keys.
+
+        Args:
+            list_: List of values
+
+        Returns:
+            dict: dictionary built
+        """
+
+        ret = {}
+
+        if not list_:
+            return ret
+
+        for value in list_:
+            if isinstance(value, dict):
+                ret.update(value)
+            else:
+                ret[str(value)] = True
+
+        return ret
+
+    @staticmethod
+    def get_logger(mod_name):
+        """
+        To activate logs, setup the environment var LOGFILE
+        e.g.: export LOGFILE=/tmp/ansible-oneview.log
+
+        Args:
+            mod_name: module name
+
+        Returns: Logger instance
+        """
+
+        logger = logging.getLogger(os.path.basename(mod_name))
+        global LOGFILE
+        LOGFILE = os.environ.get('LOGFILE')
+        if not LOGFILE:
+            logger.addHandler(logging.NullHandler())
+        else:
+            logging.basicConfig(level=logging.DEBUG, datefmt='%Y-%m-%d %H:%M:%S',
+                                format='%(asctime)s %(levelname)s %(name)s %(message)s',
+                                filename=LOGFILE, filemode='a')
+        return logger
+
+
+class ResourceComparator():
+    MSG_DIFF_AT_KEY = 'Difference found at key \'{0}\'. '
+
+    @staticmethod
+    def compare(first_resource, second_resource):
+        """
+        Recursively compares dictionary contents equivalence, ignoring types and elements order.
+        Particularities of the comparison:
+            - Inexistent key = None
+            - These values are considered equal: None, empty, False
+            - Lists are compared value by value after a sort, if they have same size.
+            - Each element is converted to str before the comparison.
+        Args:
+            first_resource: first dictionary
+            second_resource: second dictionary
+
+        Returns:
+            bool: True when equal, False when different.
+        """
+        resource1 = deepcopy(first_resource)
+        resource2 = deepcopy(second_resource)
```

Comment:
Is there a reason to do deepcopies here (and at the top of a few other methods)?  deepcopies are slow and it doesn't look like the data is being modified (for which you might want to have a pristine copy and a modified copy) so you can probably discard the deepcopy to improve your speed.